packages/utils/telemetry-utils/src/test/sampledTelemetryHelper.spec.ts

```diff
@@ -233,6 +234,95 @@ describe("SampledTelemetryHelper", () => {
 		helper.dispose();
 		assert.strictEqual(logger.events.length, 1);
 	});
+
+	it("Correctly returns computed duration for custom data", () => {
+		const sampling = 10;
+		const helper = new SampledTelemetryHelper(
+			{ eventName: "testEvent" },
+			logger,
+			sampling,
+			true /* includeAggregateMetrics */,
+		);
+
+		const clock = sinon.useFakeTimers();
+		const startingPoint = 50; // Arbitrary starting point.
+		let totalDuration = 0;
+		let maxDuration = Number.MIN_VALUE;
+		let minDuration = Number.MAX_VALUE;
+
+		for (let i = 0; i < sampling; i++) {
+			helper.measure(() => {
+				const currentIterationDuration = startingPoint + i;
+
+				clock.tick(currentIterationDuration);
+				totalDuration += currentIterationDuration;
+				maxDuration = Math.max(maxDuration, currentIterationDuration);
+				minDuration = Math.min(minDuration, currentIterationDuration);
+			});
+		}
+
+		clock.restore();
+
+		assert.strictEqual(logger.events.length, 1);
+		assert.strictEqual(logger.events[0].totalDuration, totalDuration);
+		assert.strictEqual(logger.events[0].averageDuration, totalDuration / sampling);
+		assert.strictEqual(logger.events[0].maxDuration, maxDuration);
+		assert.strictEqual(logger.events[0].minDuration, minDuration);
+	});
+});
+
+/**
+ * @remarks Initialized in advance to extract its keys for type checking.
+ * Arbitrary properties that can be logged with the telemetry event.
+ */
+interface TestTelemetryProperties {
+	propertyOne: number;
+	propertyTwo: number;
+	propertyThree: number;
+}
+
+describe("SampledTelemetryHelper with Custom Data", () => {
+	let logger: TestLogger;
+
+	beforeEach(() => {
+		logger = new TestLogger();
+	});
+
+	it("Correctly returns computed averages and maxes for custom data", () => {
+		const sampling = 10;
+		const initialCustomMetrics = {
+			"propertyOne": 0,
+			"propertyTwo": 0,
+			"propertyThree": 0,
+		};
+
+		const helper = new SampledTelemetryHelper<TestTelemetryProperties>(
+			{ eventName: "testEvent" },
+			logger,
+			sampling,
+			undefined,
+			undefined,
+			initialCustomMetrics,
+		);
+
+		for (let i = 0; i < sampling; i++) {
+			helper.measure((metricsTracker) => {
+				metricsTracker.incrementMetric({
+					propertyOne: 1,
+					propertyTwo: 2,
+					propertyThree: 3,
```

Comment:
Let's use different values per iteration (like we've done in other tests, using adding `i` here somewhere), so the test is a bit more realistic and we can be sure the code isn't just stamping the last value it saw, instead of the actual avgs/maxes.